test_runner/performance/test_storage_controller_scale.py

```diff
@@ -195,10 +271,44 @@ def check_memory():
     env.storage_controller.consistency_check()
     check_memory()
 
-    # Restart pageservers: this exercises the /re-attach API
-    for pageserver in env.pageservers:
-        pageserver.stop()
-        pageserver.start()
+    shard_counts = get_consistent_node_shard_counts(env, total_shards)
+    log.info(f"Shard counts before rolling restart: {shard_counts}")
+
+    assert_consistent_balanced_attachments(env, total_shards)
+
+    # Restart pageservers gracefully: this exercises the /re-attach pageserver API
+    # and the storage controller drain and fill API
+    for ps in env.pageservers:
+        env.storage_controller.retryable_node_operation(
```

Comment:
How come this needs the retryable_node_operation wrapper?  Can we make it more specific for which kinds of error it's meant to retry?

Reply:
It doesn't need it in this controlled environment. I followed the RFC specification for how deployment orchestrator should do.